site/documentation/content/api/management/device-registry-v1.yaml

```diff
@@ -1065,6 +1065,13 @@ components:
                   The protocol adapter needs to map this name to the particular service to invoke.
                   NB: This is an experimental feature and thus this property is subject to change without
                   notice.
+            "command-mapper":
```

Comment:
Having now a "mapper" and a "command-mapper" field doesn't look particularly well designed here.
We *could* make a clearer distinction here and rename the "mapper" field to something like "downstream-message-mapper" (or "telemetry-mapper", meaning both telemetry and event messages as used in some other places in Hono, although I think that would be rather confusing).
On the other hand one could think of the telemetry/event direction as the more important, default direction, so using just "mapper" would match that. So, I'm not sure here.
@sophokles73 Any opinion on this?

Reply:
Do we actually need to define two mappers? Will it be the usual case to use two different services for mapping the payload?

Reply:
@sophokles73, yes we require 2 different mappers. Mapping the downstream payload is sort of decoding the payload while mapping the upstream payload encodes the payload.
In our usecase some additional information is added to the payload which is not required when sending to an LNS (because that specific json payload is added by the lora providers)
Considering the name: I thought it would be clear enough to have a mapper for downstream and specifying it for upstream although maybe `downstream-mapper` and `upstream-mapper` might suffice as well?

Reply:
I am not sure if we are on the same page here. i am not arguing for using the same logic to map downstream and upstream payload. I am just saying that we could (should?) use the same service component for both directions which could be selected by means of using different URIs, e.g. using `http://my-mapper/map/downstream` vs. `http://my-mapper/map/upstream`. The `downstream` and `upstream` could even be appended to the generic endpoint URI (`http://my-mapper/map`) by the protocol adapter by convention.

Reply:
@sophokles73, am I correct that you are talking about the actual implementation while @calohmn was talking about the configuration part?
The implementation is a bit different because it has another type of context and the result is also slightly handled differently. As for the configuration: I would rather have 2 different configuration settings instead of using convention. Generating the commands might be a computational intensive operation...

Reply:
I see.
Based on the assumption that the whole *mapping functionality* is still to be considered experimental, I think that we should indeed rename the *mapper* property to *telemetry-mapper* then.

Reply:
Looking at the current implementation in the MQTT adapter, the mapper also gets applied to *command response* messages. I think that also makes sense. With respect to that, `downstream-message-mapper` would fit better.

Reply:
If we no longer use the message semantics for the names, then the *command-mapper* should probably be named *upstream-message-mapper* accordingly, right?
I am OK with both approaches, I just do not want to mix up both concepts, i.e. we either use a mapper *per direction of message flow* or we use mappers *per message type/semantics*.
Does that make sense @calohmn @BobClaerhout ?

Reply:
Ok, I'm ok with `downstream-message-mapper` and `upstream-message-mapper`. I think this is descriptive.

Reply:
I'm ok with that as well.

Reply:
I changed it and adopted the documentation accordingly